### DISCUSSION TITLE
Setting Canonical Doesn't Work in generateMetaData

### DISCUSSION THREAD

**Discussion by darrylmorley** [vercel/next.js#52783] [Help]
### Summary

I'm trying to set the canonical for my pages. As per the docs I am using the below. I have set the metadataBase in my root layout:

```
export async function generateMetadata({ params }) {
  let { slug } = params;

return {
   alternates: {
        canonical: `${slug}`,
      },
}}
```

**Comment by hyjoong** [comment]
"Slug" is the folder name you assumed, and for instance, if you structured your folder as https://test.com/item/3, then the slug would be "itemId". So, in the params object, it will be in the format `{ itemId: '3' }`. 

```javascript
export async function generateMetadata({ params }) {
console.log('params: ',params)  // params: { itemId: 3} 

return {
   alternates: {
        canonical: `${params.itemId}`,
      },
}}
```
